bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -236,4 +239,22 @@ private void cleanPropertyBuilders(final ClassModelBuilder<?> classModelBuilder)
             classModelBuilder.removeProperty(propertyName);
         }
     }
+
+    private <T> void processBsonExtraElementsAnnotation(final PropertyModelBuilder<T> propertyModelBuilder) {
+        PropertyAccessor<T> propertyAccessor = propertyModelBuilder.getPropertyAccessor();
+        if (!(propertyAccessor instanceof PropertyAccessorImpl)) {
```

Comment:
Yes, it also applies to `USE_GETTER_AS_SETTER_CONVENTION` and `SET_PRIVATE_FIELDS_CONVENTION`.

The only reason its needed is because `PropertyAccessorImpl` has the `PropertyMetadata` which has access to the fields / methods. The PropertyMetadata was purposely kept private / an implementation detail but doing so has hinders users.